### DISCUSSION TITLE
How to define middleware in layouts component?

### DISCUSSION THREAD

**Discussion by gitamgadtaula** [nuxt/nuxt#19510] [Questions]
HI, I am an old user of Nuxt2. And I have been using middlewares in layout components like 
`<script>
export default {
   middleware:'auth'
}
</script>`

Since I cannot use `definePageMeta` in layout components in Nuxt3, how can assign middleware to any layout component? For example I want to protect my default.vue layout with 'auth' middleware.

**Comment by imlautaro** [comment]
I think in Nuxt 3 you can't add a middleware on a layout, but you could create a global middleware to match multiple routes (something like `/admin/**`) or set that middleware in one parent page and it will be executed on nested routes too.

**Reply by gitamgadtaula** [reply]
Oh thats a shame. But if a Nuxt team decided to drop it, it must be for good reasons, thanks.

**Reply by gitamgadtaula** [reply]
@danielroe can you confirm please 😊?
